app/models/collection_extra.rb

```diff
@@ -26,7 +26,8 @@ def self.load
       extra.has_browseable_granules = false unless extra.has_granules
 
       if extra.has_granules
-        if extra.has_browseable_granules.nil? || (extra.has_browseable_granules && extra.browseable_granule.nil?)
+        # Edited the condition to update browseable_granule ID if it hasn't been updated for a week.
+        if extra.has_browseable_granules.nil? || (extra.has_browseable_granules && extra.updated_at < 1.week.ago)
```

Comment:
If we are grabbing the oldest granule, this may break for NRT granules, which cycle out of existence. Can you make sure we're grabbing the newest granule?